### ISSUE TITLE
No gain change

### ISSUE THREAD

**Issue by CHCHcc** [D-Winker/TSL2591_AutoAdjust#1]
I tried your AutoAdjust code but I always end up with gain config 3. It updates immediately after start from 2 to 3 and doesn't step down again when it gets brighter. So it reads "Too bright!" instead.

**Comment by D-Winker** [comment]
Thank you for the feedback and motivation! It's been a while since I worked on this code, and I wasn't able to reproduce your exact issue, but I came across many other ones. **The short answer is**: I've made some changes and pushed a new version; let me know if it works for you.  
  
The longer answer is: I tried this out indoors at night. In ambient light, the sensor settled at setting 3. I used my phone's flashlight to test the lower settings, and I covered the TSL with my hand to test the higher settings.    
I ran across a handful of issues mentioned in the commit - an infinite loop in bright light, and oscillating back and forth between settings 3 and 4. I created a new fourth gain setting and pushed the old one up to become the fifth, then it oscillated between 4 and 5 - I put in a special case to handle small values (< 0.1) with setting 4. Logically under those conditions, you would want higher gain, but something is going wrong and that small value is resulting from too much light (so we need lower gain).  
   
I think it comes down to two issues: the first is that this sensor is strange and the lux calculation is (if I recall correctly) reliant on proprietary magic numbers. The equation is a bit of a black box. We just have to work around this with special cases and parameter tuning.  
The second issue is that my parameters probably aren't well tuned. It would be good to capture data with all settings for various different brightnesses, then based on that data, calculate the appropriate conditions for moving from one setting to another.  
  
All of that being said, it's working pretty well right now.
